apps/examples/ble_rmc/ble_rmc_main.c

```diff
@@ -192,6 +190,110 @@ static uint8_t g_adv_resp[] = {
 	0x11, 0x09, 'T', 'I', 'Z', 'E', 'N', 'R', 'T', ' ', 'T', 'E', 'S', 'T', '(', '0', '2', ')',
 };
 
+static ble_scan_callback_list scan_config = {
+	ble_scan_state_changed_cb,
+	ble_device_scanned_cb,
+};
+
+static ble_client_callback_list client_config = {
+	ble_device_disconnected_cb,
+	ble_device_connected_cb,
+	ble_operation_notification_cb,
+};
+
+static ble_server_init_config server_config = {
+	ble_server_connected_cb,
+	true,
+	gatt_profile, sizeof(gatt_profile) / sizeof(ble_server_gatt_t)};
+
+static ble_result_e ble_connect_common(bool is_auto)
+{
+	ble_result_e ret = BLE_MANAGER_FAIL;
+	struct timespec abstime;
+	ble_client_state_e cli_state = BLE_CLIENT_NONE;
+	ble_conn_info conn_info = { 0, };
+
+	memcpy(conn_info.addr.mac, g_target.mac, BLE_BD_ADDR_MAX_LEN);
+	RMC_LOG(RMC_CLIENT_TAG, "BLE mac type : %d\n", g_target.type);
+	conn_info.addr.type = g_target.type;
+	conn_info.conn_interval = 8;
+	conn_info.slave_latency = 128;
+	conn_info.mtu = 240;
+	conn_info.scan_timeout = 1000;
+	conn_info.is_secured_connect = true;
+
+	sem_init(&g_conn_sem, 0, 0);
+
+	if (g_ctx == NULL) {
+		g_ctx = ble_client_create_ctx(&client_config);
+	}
+	if (g_ctx == NULL) {
+		RMC_LOG(RMC_CLIENT_TAG, "create ctx fail\n");
+		return 0;
+	} else {
+		RMC_LOG(RMC_CLIENT_TAG, "create ctx ok[%p]\n", g_ctx);
+	}
+
+	ret = ble_client_autoconnect(g_ctx, is_auto);
+	if (ret != BLE_MANAGER_SUCCESS) {
+		RMC_LOG(RMC_CLIENT_TAG, "fail to set autoconnect=%d [%d]\n", is_auto, ret);
+	}
+
+	cli_state = ble_client_get_state(g_ctx);
+	RMC_LOG(RMC_CLIENT_TAG, "Client State [ %s ]\n", __client_state_str(cli_state));
+
+	ret = ble_client_connect(g_ctx, &conn_info);
+	if (ret != BLE_MANAGER_SUCCESS) {
+		RMC_LOG(RMC_CLIENT_TAG, "connect fail[%d]\n", ret);
```

Comment:
Is it ok to continue to excute below codes after each failures?
It not, please add exit codes or conditions for executing next operations.
```
if (ret != BLE_MANAGER_SUCCESS) {
   RMC_LOG(RMC_CLIENT_TAG, "connect fail[%d]\n", ret);
   return X;
}
```
or
```
if (ret == BLE_MANAGER_SUCCESS) {
   cli_state = ble_client_get_state(g_ctx);
   RMC_LOG(RMC_CLIENT_TAG, "Client State [ %s ]\n", __client_state_str(cli_state));
    ...
}
```

Reply:
Ok, I will fix it